test/hotspot/jtreg/runtime/ConstantPool/BadMethodHandles.java

```diff
@@ -23,188 +23,153 @@
 
 /*
  * @test
+ * @enablePreview
  * @bug 8087223 8195650
  * @summary Adding constantTag to keep method call consistent with it.
- * @modules java.base/jdk.internal.org.objectweb.asm
- *          java.base/jdk.internal.misc
+ * @modules java.base/jdk.internal.misc
  *          java.management
  * @compile -XDignore.symbol.file BadMethodHandles.java
  * @run main/othervm BadMethodHandles
  */
 
-import jdk.internal.org.objectweb.asm.*;
 import java.io.FileOutputStream;
+import java.lang.classfile.ClassFile;
+import java.lang.constant.ClassDesc;
+import java.lang.constant.DirectMethodHandleDesc;
+import java.lang.constant.MethodHandleDesc;
+import java.lang.constant.MethodTypeDesc;
 import java.lang.reflect.InvocationTargetException;
-import java.lang.invoke.MethodHandle;
-import java.lang.invoke.MethodHandles;
-import java.lang.invoke.MethodType;
-import static jdk.internal.org.objectweb.asm.Opcodes.*;
+
+import static java.lang.classfile.ClassFile.*;
+import static java.lang.constant.ConstantDescs.*;
 
 public class BadMethodHandles {
 
+    private static final ClassDesc CD_System = ClassDesc.of("java.lang.System");
+    private static final ClassDesc CD_PrintStream = ClassDesc.of("java.io.PrintStream");
+    private static final ClassDesc CD_LPrintStream = ClassDesc.ofDescriptor("Ljava/io/PrintStream;");
```

Comment:
This is the same as `CD_PrintStream`, so no need for two constants.

Reply:
Thank you, changes applied.